packages/react-router/src/router.ts

```diff
@@ -1117,16 +1117,7 @@ export class Router<
       } = {},
       matches?: Array<MakeRouteMatch<TRouteTree>>,
     ): ParsedLocation => {
-      // if the router is loading the previous location is what
-      // we should use because the old matches are still around
-      // and the latest location has already been updated.
-      // If the router is not loading we should always use the
-      // latest location because resolvedLocation can lag behind
-      // and the new matches could already render
-      const currentLocation = this.state.isLoading
-        ? this.state.resolvedLocation
-        : this.latestLocation
```

Comment:
This change made the kitchensink app work - but dont know what other consequence it will have? maybe my change is not the right solution - but it solves the kitchensink app navigate from invoices to users